src/components/ec-multiple-values-selection/ec-multiple-values-selection.vue

```diff
@@ -110,120 +110,111 @@
-      type: Array,
-      required: false,
-      default: () => ([]),
-    },
-    isLoading: {
-      type: Boolean,
-      default: false,
-    },
-    errorMessage: {
-      type: String,
-    },
-    errorIcon: {
-      type: String,
-      default: 'simple-error',
-    },
-    emptyMessage: {
-      type: String,
-    },
-    emptyIcon: {
-      type: String,
-      default: 'simple-error',
-    },
-    isSearchable: {
-      type: Boolean,
-    },
-    isSelectAll: {
-      type: Boolean,
-      default: false,
-    },
-    selectAllFiltersText: {
-      type: String,
-      default: 'Select all',
-    },
-    searchFilterPlaceholder: {
-      type: String,
-      default: 'Search...',
-    },
+  emptyMessage: {
+    type: String,
   },
-  emits: ['update:modelValue', 'search', 'change'],
-  data() {
-    return {
-      query: '',
-    };
+  emptyIcon: {
+    type: String,
+    default: 'simple-error',
   },
-  computed: {
-    selectedValues() {
-      return new Set(this.modelValue.map(item => item.value));
-    },
-    allFiltersAreSelected() {
-      return this.modelValue.length === this.items.length;
-    },
-    atLeastOneFilterIsSelected() {
-      return !this.allFiltersAreSelected && this.selectedValues.size > 0;
-    },
-    searchModel: {
-      get() {
-        return this.query;
-      },
-      set(value) {
-        this.query = value;
-        this.$emit('search', value);
-      },
-    },
+  isSearchable: {
+    type: Boolean,
   },
-  methods: {
-    toggleAll() {
-      if (this.allFiltersAreSelected) {
-        this.update([]);
-      } else {
-        this.update(this.items);
-      }
-    },
-    onSelect(item) {
-      const newItems = [...this.modelValue, item];
-      this.update(newItems);
-    },
-    onDeselect(item) {
-      const newItems = this.modelValue.filter(selectedItem => selectedItem.value !== item.value);
-      this.update(newItems);
-    },
-    focus() {
-      /* istanbul ignore else */
-      if (this.isSearchable) {
-        this.$nextTick(() => {
-          this.$refs.searchInput?.focus();
-        });
-      }
-    },
-    isItemChecked(item) {
-      return this.selectedValues.has(item.value);
-    },
-    update(items) {
-      this.$emit('update:modelValue', items);
-      this.$emit('change', items);
-    },
+  isSelectAll: {
+    type: Boolean,
+    default: false,
   },
-};
+  selectAllFiltersText: {
+    type: String,
+    default: 'Select all',
+  },
+  searchFilterPlaceholder: {
+    type: String,
+    default: 'Search...',
+  },
+});
+const emit = defineEmits(['update:modelValue', 'search', 'change']);
+
+// search
+let searchQuery = '';
+const searchModel = computed({
+  get() {
+    return searchQuery;
+  },
+  set(value) {
+    searchQuery = value;
+    emit('search', value);
+  },
+});
+
+const searchInputRef = ref(null);
+function focus() {
+  if (props.isSearchable) {
+    nextTick(() => {
+      searchInputRef?.value.focus();
+    });
+  }
+}
+defineExpose({ focus });
```

Comment:
Why do we expose focus here?

Reply:
It is used in the parent component: EcSyncMultipleValuesFilter, without exposing it, the tests in the parent are failing. But let me know if you think there's a better approach